### PR TITLE
CLOUDP-90457 Create kubelinter-check.yml

### PR DESCRIPTION
**Pull request by killian2k**
Added the kubelinter action + reduce the CPU request when doing the e2e tests. Now, the manager.yaml and the openshift operator are in the mongodb namespace

### All Submissions:

* [x] Have you opened an Issue before filing this PR?
* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Have you signed all of your commits?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: .github/workflows/kubelinter-check.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+#
+# Docs: https://github.com/marketplace/actions/close-stale-issues
```
Comment (bznein) [inline .github/workflows/kubelinter-check.yml:3]: non relevant comment

File: config/manager/manager.yaml [hunk 2]
```diff
@@ -39,4 +58,15 @@ spec:
         image: quay.io/mongodb/mongodb-kubernetes-operator:0.6.2
         imagePullPolicy: Always
         name: mongodb-kubernetes-operator
+        resources:
+          limits:
+            cpu: 1100m
+            memory: 1Gi
+          requests:
+            cpu: 500m
+            memory: 200Mi
```
Comment (bznein) [inline config/manager/manager.yaml:67]: [q] How did we derive these values?

File: deploy/clusterwide/role.yaml [hunk 3]
```diff
@@ -6,10 +6,10 @@ rules:
 - apiGroups:
```
Comment (bznein) [inline deploy/clusterwide/role.yaml:6]: The diff on this file is a bit noisy, not your fault. Can you summarize what changed? I see we completely removed permissions on secrets but is there a specific reason we keep secrets in with `verbs: []`?

File: deploy/e2e/role.yaml [hunk 4]
```diff
@@ -104,3 +104,4 @@ rules:
   - patch
   - update
   - watch
+
```
Comment (bznein) [inline deploy/e2e/role.yaml:107]: [micro-nit] Can we make github happy here by just restoring this file as it was?

File: test/e2e/setup/setup.go [hunk 5]
```diff
@@ -238,6 +240,22 @@ func createOrUpdate(obj client.Object) error {
 	return nil
 }
 
+// withCPURequest assumes that the underlying type is an appsv1.Deployment.
```
Comment (bznein) [inline test/e2e/setup/setup.go:243]: Can we maybe log something in case this is not a deployment? Or maybe even fail loudly so that is clear that this function has been called on the wrong type?

**Review by bznein — COMMENTED** [review]
Overall looks good and I look forward to seeing this in action, just a few comments!

File: config/manager/manager.yaml (hunk 2 above)
Comment (killian2k) [reply config/manager/manager.yaml:67]: According to Cian these are the values used for the enterprise operator

Comment (bznein) [reply config/manager/manager.yaml:67]: makes sense, I guess at some point we'd want to perform some actual testing of the community operator to come with a precise number, but definitely out of scope here

File: deploy/clusterwide/role.yaml (hunk 3 above)
Comment (killian2k) [reply deploy/clusterwide/role.yaml:6]: Removed  create for pods, Removed everything from secret

File: test/e2e/setup/setup.go (hunk 5 above)
Comment (killian2k) [reply test/e2e/setup/setup.go:245]: Good idea, I added the error management to the options functions

File: .github/config_files/config_lint.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+checks:
+  addAllBuiltIn: true
+
+#Reasons to exclude:
+  # non-existent-service-account because the service account is created in another file
+  # minimum-three-replicas because the deployment contains only 1 replica of the operator
+  # no-readiness-probe & no-liveness-probe because for now, it brings nothing to add theses probes
```
Comment (chatton) [inline .github/config_files/config_lint.yaml:7]:
```suggestion
  # no-readiness-probe & no-liveness-probe because for now, it brings nothing to add these probes
```

File: .github/workflows/kubelinter-check.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
+name: Kubelinter-check
+
+on:
+  push:
+    branches:
+    - master
+    paths-ignore: 
+    - docs/**
```
Comment (chatton) [inline .github/workflows/kubelinter-check.yml:8]: I think maybe we don't need need to exclude any directories, it's possible that we could have yaml samples under `docs`

File: deploy/e2e/service_account.yaml [hunk 8]
```diff
@@ -2,3 +2,4 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: e2e-test
+  namespace: default
```
Comment (chatton) [inline deploy/e2e/service_account.yaml:5]: can we add a note here saying why we are still using the default ns for tests (TLS fixtures assume default ns)

**Review by chatton — COMMENTED** [review]
This looks great! I just had a few small comments but I think we're good to merge afterwards.

**Review by bznein — COMMENTED** [review]
LGTM for what concerns my comments :) Leaving to Cian final LGTM since I see he has some comments

**Review by chatton — APPROVED** [review]
LGTM! Nice work on this!
